library/common/http/dispatcher.cc

```diff
@@ -31,6 +31,13 @@ Dispatcher::DirectStreamCallbacks::DirectStreamCallbacks(DirectStream& direct_st
 void Dispatcher::DirectStreamCallbacks::encodeHeaders(const HeaderMap& headers, bool end_stream) {
   ENVOY_LOG(debug, "[S{}] response headers for stream (end_stream={}):\n{}",
             direct_stream_.stream_handle_, end_stream, headers);
+
+  // @see Dispatcher::resetStream's comment on its call to runResetCallbacks.
+  // Envoy only deferredDeletes the ActiveStream if it was fully closed otherwise it issues a reset.
+  if (direct_stream_.local_closed_ && end_stream) {
+    direct_stream_.pending_stream_deferred_delete_ = true;
```

Comment:
nit: `pending_cleanup` or `pending_delete` is a little more concise and feels equally expressive in this case. definitely nabd.

Reply:
@goaway I wanted to make it clear that it isn't the direct_stream that is pending deletion, but the HCM's active_stream that the direct_stream shadows. I am concerned that `direct_stream_.pending_cleanup` makes it seem like the direct stream is the one pending cleanup.

Perhaps `hcm_stream_pending_destroy`?